### ISSUE TITLE
Storage add-on: A user can't buy the 50G add-on and then buy the 100G add-on

### ISSUE THREAD

**Issue by southp** [Automattic/wp-calypso#86150]
### Quick summary

_This issue is broken out from the auditing in https://github.com/Automattic/wp-calypso/issues/85773._

![Image](https://github.com/Automattic/wp-calypso/assets/1842898/b117b6e8-93f4-42de-8170-10627c90811d)

As shown here, my test Business site owning a 50G add-on can't buy the 100G add-on since it's absent. In /plans, it's even more confusing since I can select it but it will show me "Contact Support"

![Image](https://github.com/Automattic/wp-calypso/assets/1842898/f5c216a0-432f-4a9b-9507-38b3caf1ad9f)

It should be available since the maximum storage is 200G. From a quick glance of the code, the way `exceedsSiteStorageLimits` computed is wrong.

### Steps to reproduce

1. Create a test site on the Creator(Business) plan.
2. Purchase the 50G add-on
3. Access `/add-ons`

### What you expected to happen

I'd see the 100G add-on available.

### What actually happened

The 100G add-on doesn't show.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable.

**Comment by rickmgithub** [comment]
## 📌 REPRODUCTION RESULTS ##
Replicated exactly as above.

## 📌 ACTIONS ##
Triaged

**Comment by chriskmnds** [comment]
@southp as far as I checked this will also fail on checkout, so it's actually correctly removed in the UI. Either the site cannot upgrade to another 100GB or only a single storage add-on purchase is permissible. 

I think/recall part of my testing the difference between the 200GB limit and a Creator site with extra 50GB was 93 and not 100. Not sure where the other 7 came from - which would make for additional 100GB purchase violate the 200 limit. We need to check the backend.

**Comment by inaikem** [comment]
@mmtr - tagging you for action per our Group Triage session.

**Comment by mmtr** [comment]
I think the issue is a bit different now, and it depends on whether the site is Simple or Atomic, and whether the admin interface is set to Default or Classic.

**Simple sites – Default admin interface**

- Can I purchase storage from the Plans page? ❌ 
- Is the "Upgrades > Add-ons" menu available? ✅ 
- Can I purchase storage from the Add-ons page? ✅ 
- The "Needs more storage?" link in the Hosting overview page opens the Plans page, where I cannot purchase extra storage ❌ 

Plans page | Add-ons page | Hosting overview page
--- | --- | ---
![Image](https://github.com/user-attachments/assets/626ce59f-806a-4ddb-b8c7-342add193742) | ![Image](https://github.com/user-attachments/assets/e6fb6c41-dc95-4091-a86c-d2ce6dabaa18) | ![Image](https://github.com/user-attachments/assets/35526e89-15dd-4bcd-93d8-2a1913a0111d)

**Simple sites – Classic admin interface**

- Can I purchase storage from the Plans page? ❌ 
- Is the "Hosting > Add-ons" menu available? ✅ 
- Can I purchase storage from the Add-ons page? ✅ 
- The "Needs more storage?" link in the Hosting overview page opens the Plans page, where I cannot purchase extra storage ❌ 

Plans page | Add-ons page | Hosting overview page
--- | --- | ---
![Image](https://github.com/user-attachments/assets/5103ef44-d3ca-4df0-89e3-07157fa6c24f) |![Image](https://github.com/user-attachments/assets/2830a6cc-bd52-4a41-9a68-cb37180881b1) | ![Image](https://github.com/user-attachments/assets/8a5419c9-83d2-4432-a343-4a0cefd3a372)

**Atomic sites – Default admin interface**

- Can I purchase storage from the Plans page? ✅ 
- Is the "Upgrades > Add-ons" menu available? ❌ 
- Can I purchase storage from the Add-ons page? ✅ 
- The "Needs more storage?" link in the Hosting overview page opens the Add-ons page, which is not accessible from any menu (but I can at least purchase extra storage here) ⚠ 

Plans page | Add-ons page | Hosting overview page
--- | --- | ---
![Image](https://github.com/user-attachments/assets/b361f3b4-655b-4be8-9ff6-abca63865483) | ![Image](https://github.com/user-attachments/assets/d0500548-6a48-415a-82b2-fe91c798de71) | ![Image](https://github.com/user-attachments/assets/8d4b52fd-47f8-4b96-a9d2-1459a3397fac)

**Atomic sites – Classic admin interface**

- Can I purchase storage from the Plans page? ✅ 
- Is the "Hosting > Add-ons" menu available? ✅ 
- Can I purchase storage from the Add-ons page? ✅ 
- The "Needs more storage?" link in the Hosting overview page opens the Add-ons page, where I cannot purchase extra storage ✅  

Plans page | Add-ons page | Hosting overview page
--- | --- | ---
![Image](https://github.com/user-attachments/assets/d7fc5c5b-48c0-48ae-8312-336c6069f358) | ![Image](https://github.com/user-attachments/assets/88949f47-c9b7-4c47-96f5-00595ca61c4d) | ![Image](https://github.com/user-attachments/assets/bb52f95f-878c-428f-9cb5-3eb4d470ea5c)

The experience is quite inconsistent 🤯 

Ideally:
- Extra storage should be purchased in the same places (Plans, or Add-ons, or both) across all scenarios
- The "Needs more storage?" link should like to the same page always
- The Add-ons menu should be visible at all cases

cc @Automattic/martech @Automattic/lego

**Comment by BogdanUngureanu** [comment]
Opened D164593-code and https://github.com/Automattic/wp-calypso/pull/95678 which fixes this issue.

I also opened a nice to have change in /plans page that adds the addon dropdown for Personal and Premium plans: https://github.com/Automattic/wp-calypso/pull/95677#issuecomment-2435567725 - we can skip this one if we want.

**Comment by mmtr** [comment]
@BogdanUngureanu I'm moving this back to "In Progress" since I think some more changes are still needed for D164593-code to fix the tests. 

It also seems that https://github.com/Automattic/wp-calypso/pull/95677 should be abandoned?

Feel free to change the status to "Needs Review" once everything is ready.
